packages/x-date-pickers/src/locales/index.ts

```diff
@@ -5,6 +5,7 @@ export * from './deDE';
 export * from './elGR';
 export * from './enUS';
 export * from './esES';
+export * from './euES';
```

Comment:
Should we shorten it this? e.g. https://github.com/Chocobozzz/PeerTube/issues/3956

```suggestion
export * from './eu';
```

Reply:
Thank you. That's a very good point and valuable discussion
@lajtomekadimon Could you confirm your position on this topic? Do you agree that having only `eu` as a locale name makes sense?

Reply:
Both "eu" and "eu-ES" are fine as identifiers for the language, but I don't think `eu` alone is a good idea in our case. `euES` is the right choice here, mainly because that's the same criteria applied for the other languages.

All locale codes of `packages/x-date-pickers/src/locales/` have both the language code and the country code. You have `enUS` instead of `en` for English, `esES` instead of `es` for Spanish, etc. Even languages like Icelandic or Japanese, which are only spoken on one country (at least to my knowledge), are `isIS` and `jaJP`.

I can't see the reason why Basque must be an exception and has a neutral country code.

And if that is not enough reason, keep in mind that a locale represents regional settings. It's much more than a language: it includes how numbers, dates and times are display, among other things.

Reply:
Yes, I've read it. But that makes sense only if you apply the same criteria to all languages.

I'm not sure if changing the policy of all locale codes is related to this PR. I think that should be addressed on its own issue, since that affects all languages, right?

Reply:
Just my two cents on this.

Unlike other minoritarian languages in Spain that might be considered "dialects" very close to Spanish Basque is a totally different language. And in fact there are many dialects of Basque although there is what is called "Euskara Batua" (Unified Basque) which is a kind of "formal" or ""official"" Basque. That's not the Basque that you will hear in the streets of many town, but it's the Basque that you should read in most official or institutional documents.

The localization that we are providing is, if not completely in "Batua", probably quite close to it... I hope. So yes, probably could be considered as a "middleground" common Basque and consider it "country neutral". But to be honest, not sure if it will feel very natural for Basque speakers in France.

In any case I think being a minoritarian language and not having provided any alternative "eu_FR" locale I think that any of both options, using just "eu" or "eu_ES" might be a good improvement.

As already commented I think that the choice of one or the other seems better related to maintaining consistency with the rest of your locales.

Reply:
Thank you both for the replies and your opinions! 🙏 


> `euES` is the right choice here, mainly because that's the same criteria applied for the other languages.

You are correct in this regard, but on the other hand, we probably didn't take enough time to look into how to handle l10n introduction for various locales properly. We might just have accepted the PR in the form that the authors created it in. 🙈 

> I can't see the reason why Basque must be an exception and have a neutral country code.

I've looked at it from a different angle—should this locale be the first one to trigger a discussion about whether we are handling l10n correctly?
We rely on the various date libraries to handle the date formatting in a localized manner and we could draw inspiration from them as well. 🤔 
Here is the Basque (`eu`) locale for the various date libraries we support:
- https://github.com/date-fns/date-fns/blob/main/src/locale/eu/index.ts
- https://github.com/iamkun/dayjs/blob/dev/src/locale/eu.js
- https://github.com/moment/moment/blob/develop/locale/eu.js

IMHO, this approach of using only the language code, where it makes sense (French, German, Estonian, Basque, etc.) makes sense to me.
For date libraries, it might make sense in certain cases to have different language and country code specific locales, when the date formats differ between countries.
But in the case of our libraries, we only handle the localization of the text used in various places. I fail to imagine a case, where our l10n would end up being country dependent (given there is no difference in the language itself).



> And if that is not enough reason, keep in mind that a locale represents regional settings. It's much more than a language: it includes how numbers, dates and times are display, among other things.

IMHO, in this case, we are starting to mix the concepts up.
How I see it, you are talking about `i18n` and indeed, that encompasses everything you mention here.
But especially in the case of our libraries, it's strictly `l10n`. Hence, no country specific formatting is concerned.

We also have a new example PR that also raises very similar questions: https://github.com/mui/mui-x/pull/10935

## My proposal 

- Let this locale be the catalyst for making our locales more correct and go with `eu`;
- Accept the mentioned Macedonian PR with `mk` code as well;
- Remove the extraneous information from the existing locales, where it serves no purpose (e.g.: English, Estonian, Finish, French, German, etc.) on v7 (`enUS` -> `en`, `fiFI` -> `fi`);

We would get the added benefit of avoiding the need to map the date adapter locales into ours in all or most cases (if need be).

P.S. In order to execute this correctly, we'd need agreement from all the localized products (currently: Material, Data Grid and Pickers).

WDYT about such a plan @lajtomekadimon @igorbga @oliviertassinari @flaviendelangle?

Reply:
For english, couldn't we imagine people wanted an `enUK` locale with the UK spellings (color vs colour for example)?

Reply:
By no means, we would and should accept such locales if someone provides them. 👍 
Date libraries also handle it in the same way, there is the `en` (en-US) locale and any other different country specific English locales (i.e. en-GB, en-IE, en-AU, etc.).

Reply:
First, I would prefer we get this one "right", so we don't introduce breaking changes later.

Second, https://cldr.unicode.org/index/downloads is I believe the source of truth used by Linux, macOS, Windows, it has a tone of locales, in the file I downloaded, I found 1044 locales, and only one eu (these are aliases)

![Screenshot 2023-11-08 at 22 47 56](https://github.com/mui/mui-x/assets/3165635/4b3b7b71-6fb6-486f-8e54-aa4f276103b8)

Third, I think it boils down to how many locale variations is there?

A. If there is only one, then I think `eu` makes more sense because it would be exceedingly specific otherwise.
B. If there is more than one, then `eu-ES` makes more sense because we signal to folks what it targets.

Part of the problem is that how many locales we might support is a moving target. I can also see how shorting to only language without the country makes sense when it's implicit enough, but it's extra complexity as it means we need to introduce a new concept: **aliases**.

Maybe we should stick to the simple concept of language-region and have rare exceptions where we know we will never go more granular?

Reply:
I agree with the idea of using just the language code, but it seems you don't have a clear policy to manage locales yet.

Correct me if I'm wrong, but the main proposal here is to use `eu` and change the other locales later in order to prevent compatibility issues, right?

Should I refactor `euES` to `eu` now, or wait until you define your policy?

I think the aliases proposal of @oliviertassinari is interesting. We could have _both_, choosing one as the reference of the other. If we go this way, we can reorganize locales without breaking changes. But there are different ways to do this:

1. Extending language-only locales, so each locale can inherit the "default translations" of the language, and optionally rewrite them.
2. Choosing one language-and-country locale as the default; for example: `en` as an alias of `enUS`, `es` as an alias of `esES`, etc.

If you ask me, I'd go with the first option. I'm thinking what happens with artificial languages like Esperanto, for example, where you don't have a direct relation with a particular country, and language-only codes seem better to me.

Reply:
> 1. First, I think we should get this one "right" here, so we don't introduce breaking changes later.

I totally agree.

I also like the idea of "aliases". It's probably less trivial to do in the realm of JS and we'd just probably end up spreading the root locale into country specific locale and overriding only what's needed. 🤔 
But, as mentioned, the alias idea could allow us to provide a migration path without breaking changes on locale names. 💯 💡 


> Maybe we should stick to the simple concept of `language-region` and have rare exceptions where we know we will never go more granular with only `language`?

At first look, it doesn't seem like a rare occasion. There is probably around at least a third of the locales that can have their country specific suffix removed. 🙈 


___



> Correct me if I'm wrong, but the main proposal here is to use `eu` and change the other locales later in order to prevent compatibility issues, right?

Of course. We're just trying to decide the path forward here. If we decide, to keep the existing `locale-COUNTRY` format, then there is nothing we need to do with this PR.
Otherwise, this would be the first locale without the country suffix and the following refactoring would be done separately. 👌

Reply:
I have updated the PR with the ideas we've discussed.
Let me know what you think @oliviertassinari @lajtomekadimon 🙏

Reply:
To me seems fine. :)

Reply:
> At first look, it doesn't seem like a rare occasion. There is probably around at least a third of the locales that can have their country specific suffix removed. 🙈

@LukasTy I wonder if there are this many, e.g. `ja` yes, `en` no, `fr` no, `es` no, `de` no.

Reply:
> WDYT? 🤔

@LukasTy Alright, here is my clearer take on it. The root of the problem here is this:

![basque-mapS](https://github.com/mui/mui-x/assets/3165635/65d11713-8dc3-46ff-ac52-b08782e688c0)

a. If we use `eu_ES` and `eu_FR` it would be confusing because these are the same
b. If we only use `eu_ES`, it could be seen as unfair for people living in France
c. If we only use `eu`, it could be seen as an exception to the rule

Overall, I think c. is the way, under the rule that we precise locales as much as possible but without being too precise when it could lead to confusion. This case is rare.

---

Then there are other cases:

2. Languages that only exist in one locale, e.g. ja-JP. But https://github.com/iamkun/dayjs/blob/dev/src/locale/ja.js feels wrong, being more specific is clearer.
3. Languages in different locales, e.g. en-UK, en-US. But https://github.com/iamkun/dayjs/blob/dev/src/locale/en.js feels wrong. It's not specific enough, I want to know exactly what `en` is. I wouldn't use this. I think that the only value of `en` is to help people who are short in time, who don't have the bandwidth to support too many locales, to know what's the best approximation (most popular locale).

My [#suggestion](https://www.notion.so/mui-org/Olivier-Tassinari-a6d0f1d19f8a4cfbb20469878ded8aa8?pvs=4#ac52aa0d7db5447b832416ef2c790f83) would go like this:

- 👍 to merge this PR
- 👍 to create en as an alias of en-US
- 👍 to extend en-US when we create en-UK
- 👍 to not touch esES, frFR, etc.

Reply:
> 2\. Languages that only exist in one locale, e.g. ja-JP. But https://github.com/iamkun/dayjs/blob/dev/src/locale/ja.js feels wrong, being more specific is clearer.

I agree about the clarity aspect, but I feel that's what the libraries have addressed by adding a comment about the exact locale specifics, like in this case: https://github.com/date-fns/date-fns/blob/main/src/locale/pt-BR/index.ts#L11
I feel that having the comments would benefit our localization files even in this state to further increase the clarity. 🤔 
This comment ties into the 3rd point as well.

Reply:
> comments

@LukasTy I think that the problem is that this is not clear from the codebase of developers. They have to open the docs or look at the IntelliSense